content/en/docs/components/training/overview.md

```diff
@@ -22,136 +24,55 @@ Training Operator implements centralized Kubernetes controller to orchestrate di
-| ------------ | ---------------------------------------------------- |
-| PyTorch      | [PyTorchJob](/docs/components/training/pytorch/)     |
-| TensorFlow   | [TFJob](/docs/components/training/tftraining/)       |
-| XGBoost      | [XGBoostJob](/docs/components/training/xgboost/)     |
-| MPI          | [MPIJob](/docs/components/training/mpi/)             |
-| PaddlePaddle | [PaddleJob](/docs/components/training/paddlepaddle/) |
-
-## Architecture
-
-This diagram shows the major features of Training Operator and supported ML frameworks.
+please follow [this guide](/docs/components/training/user-guides/mpi/) to install MPI Operator V2.
 
 <img src="/docs/components/training/images/training-operator-overview.drawio.png"
   alt="Training Operator Overview"
   class="mt-3 mb-3">
 
 Training Operator is responsible for scheduling the appropriate Kubernetes workloads to implement
-various distributed training strategies for different ML frameworks. The following examples show
-how Training Operator allows to run distributed PyTorch and TensorFlow on Kubernetes.
-
-### Distributed Training for PyTorch
-
-This diagram shows how Training Operator creates PyTorch workers for
-[ring all-reduce algorithm](https://tech.preferred.jp/en/blog/technologies-behind-distributed-deep-learning-allreduce/).
-
-<img src="/docs/components/training/images/distributed-pytorchjob.drawio.svg"
-  alt="Distributed PyTorchJob"
-  class="mt-3 mb-3">
+various distributed training strategies for different ML frameworks.
 
-User is responsible for writing a training code using native
-[PyTorch Distributed APIs](https://pytorch.org/tutorials/beginner/dist_overview.html)
-and create a PyTorchJob with required number of workers and GPUs using Training Operator Python SDK.
-Then, Training Operator creates Kubernetes pods with appropriate environment variables for the
-[`torchrun`](https://pytorch.org/docs/stable/elastic/run.html) CLI to start distributed
-PyTorch training job.
+## Why Training Operator ?
 
-At the end of the ring all-reduce algorithm gradients are synchronized
-in every worker (`g1, g2, g3, g4`) and model is trained.
+Training Operator addresses Model Training and Model Fine-Tuning step in AI/ML lifecycle as shown on
+that diagram:
 
-You can define various distributed strategies supported by PyTorch in your training code
-(e.g. [PyTorch FSDP](https://pytorch.org/docs/stable/fsdp.html)), and Training Operator will set
-the appropriate environment variables for `torchrun`.
-
-### Distributed Training for TensorFlow
-
-This diagram shows how Training Operator creates TensorFlow parameter server (PS) and workers for
-[PS distributed training](https://www.tensorflow.org/tutorials/distribute/parameter_server_training).
-
-<img src="/docs/components/training/images/distributed-tfjob.drawio.svg"
-  alt="Distributed TFJob"
+<img src="/docs/components/training/images/ml-lifecycle-training-operator.drawio.svg"
```

Comment:
I love this diagram. Based on my [comment here](https://github.com/feast-dev/feast/issues/4139), it may make sense to also feature serving along with model serving (though I note that there are some architectural choices to be had on that direction). Regardless, I think it's worth stating explicitly.

Reply:
That makes sense, do we want to update the diagram once we discuss the architecture for Feature Serving ?
For this diagram, I just took diagram that we worked together with @ronaldpetty @zanetworker for CNCF WG AI WhitePaper
https://www.cncf.io/wp-content/uploads/2024/03/cloud_native_ai24_031424a-2.pdf

Reply:
Oh nice, looks like that was very recent. @ronaldpetty @zanetworker let me know if you have any thoughts/opinions there. Would love for feature serving to be included in this as I think it is becoming increasingly more important.

Reply:
@franciscojavierarceo Not sure how to make it simple in this diagram. What do you think about this:
![Screenshot 2024-04-24 at 23 20 02](https://github.com/kubeflow/website/assets/31112157/4aeb9ae8-accc-441a-b16e-8565b1552dea)

Reply:
This may be overkill...and it's technically missing the measurement required to actually repeat the process (e.g., a click for a recommendation engine).

![Screenshot 2024-04-24 at 11 55 19 PM](https://github.com/kubeflow/website/assets/4163062/caa03699-8f23-4701-af97-916c6da5f05f)

Reply:
That makes sense, thanks @franciscojavierarceo!
A few questions:
1. Isn't the feature store also use for Model Training and Optimization ?
2. Should we split the Model Development, Iteration, and Optimization with:
Model Experimentation and Development (Kubeflow Notebooks) --> Model Optimization and Hyperparameter  Tuning (Katib) (similar to this flow: https://www.kubeflow.org/docs/started/architecture/#introducing-the-ml-workflow).
Since  I also want to use this diagram in other docs: Kubeflow Notebooks, Kubeflow Katib.

Reply:
That looks good, I would also split Model Experimentation with Chose ML Algorithms + Code Model and HP Tuning + Architecture Search.
That will allow us to explain users on which stage Kubeflow Notebooks are used on which Kubeflow Katib is used

Reply:
Will send you a message on slack

Reply:
@franciscojavierarceo this sounds like a good discussion to have on #wg-artificial-intelligence :)

If we are looking for abstractions, I'd categorize it as Data prep, model building (training, tuning, experimentation, feature engineering,...), model serving & deployment (pull from registry, deploying inference,...), then operation & eval (monitoring,..). 

I.e., there are phases in the lifecycle (building, serving, operations and iteration), components (feature stores, registries, infra,...), and personas (highlighted in the white paper, but also here: https://tag-runtime.cncf.io/wgs/cnaiwg/glossary/#data-engineers, https://tag-runtime.cncf.io/wgs/cnaiwg/glossary/#data-engineers)